modules/library/main/src/main/java/org/geotools/feature/simple/SimpleFeatureBuilder.java

```diff
@@ -361,6 +361,12 @@ public static SimpleFeature build(SimpleFeatureType type, Object[] values, Strin
      * @param id FeatureID for the generated feature, use null to allow one to be supplied for you
      */
     public static SimpleFeature build(SimpleFeatureType type, List<Object> values, String id) {
+        if (type.getAttributeCount() != values.size()) {
```

Comment:
This is a backward incompatible behavioral change, it will cause issues outside of GeoTools, and it's the reason for the importer test breaking.

The computed feature types are the expected ones (original, transformed):

```
SimpleFeatureTypeImpl AttributeComputeTransformTest:river identified extends Feature(id:id,geom:geom,river:river,flow:flow)
SimpleFeatureTypeImpl AttributeComputeTransformTest:river identified extends Feature(id:id,geom:geom,river:river,flow:flow,theDate:theDate)
```

But the code is building a new feature with the new feature type, providing a list of values from the old type (thus, it's missing one value).

The current builder can build a feature with values missing. The Javadoc did not say anything about matching the number of attributes, so there are likely implementations that provide more or fewer attributes than planned.

The change is not wrong per se mind, but if you stick to it, you'll have to change downstream code and add some instructions in the upgrade guide.

Reply:
Andrea re thought about this and decided that builder needs to be forgiving as much as possible. It already is doing type conversion and anything possible to get a valid feature out - so supplying a few default values is on brand!

Also the way addAll was written it was adding from the present "index" ... so it was just undocumented; it was always intended to allow less attributes. 

I did notice when writing test cases that builder setValidation(true) was not managing to validate individuals values resulting in another new jira ticket.